File: src/AndroidNativeBenchmark/app/src/main/java/com/rochapires/admin/androidnativebenchmark/Activities/CustomAdapter.java
package com.rochapires.admin.androidnativebenchmark.Activities;

import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.rochapires.admin.androidnativebenchmark.R;

/**
 * Created by admin on 21/06/2015.
 */
public class CustomAdapter extends BaseAdapter {

    ActivityListVisualPerformance activity;
    private static int NUMBER_OF_ROWS = 50;

    public CustomAdapter(ActivityListVisualPerformance activity) {
        this.activity = activity;
    }

    @Override
    public int getCount() {
        return NUMBER_OF_ROWS;
    }

    @Override
    public Object getItem(int position) {
        return null;
    }

    @Override
    public long getItemId(int position) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        View view;// = convertView != null ? activity.getLayoutInflater().inflate(R.layout.listItem, parent, false);

        if(convertView!= null) {
            view = convertView;
        } else {
            view = activity.getLayoutInflater().inflate(R.layout.list_item, parent, false);
        }

        TextView textView = (TextView)view.findViewById(R.id.textView);
        ImageView imageView = (ImageView)view.findViewById(R.id.imageView);
        textView.setText("elebleriii");

        imageView.setImageResource(R.drawable.profilepic);

        return view;
    }

}

File: src/AndroidXamarinBenchmark/AndroidXamarinBenchmark/Helpers/CustomAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using AndroidXamarinBenchmark.Activities;

namespace AndroidXamarinBenchmark.Helpers
{
    class CustomAdapter : BaseAdapter
    {
        //List<Contact> _contactList;
        ActivityListVisualPerformance _activity;
        private static int NUMBER_OF_ROWS = 50;

        public CustomAdapter(ActivityListVisualPerformance activity)
        {
            _activity = activity;
        }

        public override int Count
        {
            get { return NUMBER_OF_ROWS; }
        }

        public override Java.Lang.Object GetItem(int position)
        {
            return null;
        }

        public override long GetItemId(int position)
        {
            return 0;
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            var view = convertView ?? _activity.LayoutInflater.Inflate(
                Resource.Layout.ListItem, parent, false);
            var textView = view.FindViewById<TextView>(Resource.Id.textView);
            var imageView = view.FindViewById<ImageView>(Resource.Id.imageView);
            textView.Text = "elebleriii";

            //if (_contactList[position].PhotoId == null)
            //{
            //    imageView = view.FindViewById<ImageView>(Resource.Id.imageView);
            imageView.SetImageResource(Resource.Drawable.profilePic);
            //}
            //else
            //{
            //    var contactUri = ContentUris.WithAppendedId(
            //        ContactsContract.Contacts.ContentUri, _contactList[position].Id);
            //    var contactPhotoUri = Android.Net.Uri.WithAppendedPath(contactUri,
            //        Contacts.Photos.ContentDirectory);
            //    imageView.SetImageURI(contactPhotoUri);
            //}
            return view;
        }
    }
}